### ISSUE TITLE
Feature request: GetRoute

### ISSUE THREAD

**Issue by szguoxz** [johnyburd/net-route#28]
This is a very nice little package, but can we have a small feature?
GetRoute(packet)

given a packet, can you return the route information?
-----------
Or to make it simple:
GetRoute(src: SocketAddr, dest: SocketAddr)

I mean I need the route information for a packet, or a src/dst address pair with port information. As the kernel may need these information to match certain ip rules to apply certain routing tables.

**Comment by larseggert** [comment]
Yes, the ability to get the route entry towards a destination IP address would be great.
